src/charm.py

```diff
@@ -74,15 +76,19 @@ def _on_install(self, _) -> None:
             if not mysql_shell.present:
                 logger.debug(f"Installing '{MYSQL_SHELL_SNAP_NAME}' snap")
                 mysql_shell.ensure(snap.SnapState.Latest, channel="stable")
-        except snap.SnapNotFoundError as e:
-            logger.exception(f"Failed to find the '{MYSQL_SHELL_SNAP_NAME}' snap", exc_info=e)
-            self.unit.status = BlockedStatus(f"Failed to find '{MYSQL_SHELL_SNAP_NAME}'")
-            return
-        except snap.SnapError as e:
+        except (snap.SnapNotFoundError, snap.SnapError) as e:
             logger.exception(f"Failed to install the '{MYSQL_SHELL_SNAP_NAME}' snap", exc_info=e)
             self.unit.status = BlockedStatus(f"Failed to install '{MYSQL_SHELL_SNAP_NAME}'")
             return
 
+        # Update the mysql configuration - from templates/mysqld.cnf
+        try:
+            mysql_helpers = self._get_mysql_helpers()
```

Comment:
should call `mysqlsh --help` after the installation of `mysql-shell` (to create mysql shell common directories) and avoid the `MySQL._ensure_mysqlsh_common_dir()` method

Reply:
I say to let this kind of domain into the helper, so charm code deals with charming tasks.

Reply:
I don't disagree, and if we extrapolate this path, we would create another helper to deal with all installation duties.

Reply:
moved all install actions into a static method in MySQL helper class